### PR TITLE
Add examples in functional.py and loss.py

### PR DESCRIPTION
**Pull request by yunjey**
(1) functional.py

Added examples for `F.cross_entropy`, `F.binary_cross_entropy` and `F.binary_cross_entropy_with_logits`.

(2) loss.py

Added examples for `nn.BCELoss` and `nn.BCEWithLogitLoss`.

### CODE REVIEW

**Comment by yunjey** [comment]
@fmassa Here's a new pull request. Also, I have a question for PyTorch docs style.

http://pytorch.org/docs/master/nn.html#conv1d

image

As you can see from the link or the uploaded image, the `Shape` has a gray box. Is this intended?

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -286,10 +286,25 @@ class BCELoss(_WeightedLoss):
...
+          dimensions
+        - Target: :math:`(N, *)`, same shape as the input
 
+    Examples::
+
+        >>> m = nn.Sigmoid()
+        >>> loss = nn.BCELoss()
+        >>> input = autograd.Variable(torch.randn(3), requires_grad=True)
+        >>> target = autograd.Variable(torch.FloatTensor(3).random_(2))
+        >>> output = loss(m(input), target)
```
Comment (alykhantejani) [inline torch/nn/modules/loss.py]: nit: rename `m` to `sigmoid`

File: torch/nn/functional.py [hunk 2]
```diff
@@ -795,6 +809,13 @@ def binary_cross_entropy_with_logits(input, target, weight=None, size_average=Tr
                 over observations for each minibatch. However, if the field
                 sizeAverage is set to False, the losses are instead summed
                 for each minibatch. Default: True
+
```
Comment (alykhantejani) [inline torch/nn/functional.py]: Not sure we need the example in both `functional.py` and in `loss.py` as these functions link to the module docs i.e. `See :class:`~torch.nn.BCEWithLogitsLoss` for details`?

Comment (yunjey) [inline torch/nn/functional.py]: http://pytorch.org/docs/master/nn.html#torch.nn.functional.nll_loss

This work is motivated by the link above. Both `F.nll_loss` and `nn.NLLLoss` has examples. I think it would be nice to have the examples in both places to help understading.

Comment (alykhantejani) [inline torch/nn/functional.py]: @yunjey Great, thanks for the context.

File: torch/nn/modules/loss.py (hunk 1 above)
Comment (alykhantejani) [inline torch/nn/modules/loss.py]: nit:`torch.FloatTensor(3).random_(2)` could also be more concisely written as `torch.randn(3)`

**Comment by soumith** [comment]
thanks Yunjey!
